src/SWP/Bundle/ElasticSearchBundle/Repository/ArticleRepository.php

```diff
@@ -138,4 +139,35 @@ public function findByCriteria(Criteria $criteria, array $extraFields = [], bool
 
         return $this->createPaginatorAdapter($query);
     }
+
+
+    public function getSuggestedTerm(string $term): string
+    {
+        $suggestQuery = new Suggest();
+        $suggestQuery->phrase('our_suggestion', '_all');
+
+        $phraseMultiMatchQuery = new MultiMatch();
+        $phraseMultiMatchQuery->setQuery($term);
+        $phraseMultiMatchQuery->setFields('_all');
+        $phraseMultiMatchQuery->setType(MultiMatch::TYPE_PHRASE);
+        $phraseMultiMatchQuery->setParam('boost', 50);
+
+        $query = new \Elastica\Query($phraseMultiMatchQuery);
+        $suggest = new \Elastica\Suggest();
+        $suggest->setParam(
+            'phrase',
+            [
+                'text' => $term,
+                'phrase' => ['field' => '_all']
+            ]
+        );
+
+        $query->setSuggest($suggest);
+
+        $adapter = $this->createPaginatorAdapter($query);
+        $suggest = $adapter->getSuggests();
+
+        return $suggest['phrase'][0]['options'][0]['text'] ?? '';
```

Comment:
please check if keys exists before using them

Reply:
this null coalescing operator `??` does that even for nested keys. it's quite handy.